### PR TITLE
CGO fun.

### PR DESCRIPTION
**Pull request by fengttt**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by yingfeng — APPROVED** [review]

File: cgo/mo.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mo_impl.h"
+
```
Comment (JinHai-CN) [inline cgo/mo.c:18]: Where are the functions of Bitmap_Add/Bitmap_Remove/Bitmap_Contains used?

Comment (daviszhen) [reply cgo/mo.c:18]: mo.c,  bitmap.h

**Comment by fengttt** [comment]
> very interesting. any data on perf impact?

Not a problem assume vector is 4K+ long.

**Comment by nnsgmsone** [comment]
> very interesting. any data on perf impact?

I don't think batching will have a big impact, but add, set, contains are not appropriate operations for cgo.

**Comment by fengttt** [comment]
> > very interesting. any data on perf impact?
> 
> I don't think batching will have a big impact, but add, set, contains are not appropriate operations for cgo.

We should never call set/contains etc in real code.    Actually, we should never call bitmap cgo in real code.   This is just for example and test.    

I expect cgo will be used in 
1. vectorize code, and 
2. some really complex/heavy 3rd party library, for example json query jq stuff.

**Pull request merged**